src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -65,15 +65,14 @@ private string GetEquivalenceKey(SyntaxNode node)
             SyntaxEditor editor, CancellationToken cancellationToken)
         {
             // a method can have multiple `return null;` statements, but we should only fix its return type once
-            var alreadyHandled = PooledHashSet<TypeSyntax>.GetInstance();
+            using var _ = PooledHashSet<TypeSyntax>.GetInstance(out var alreadyHandled);
 
             foreach (var diagnostic in diagnostics)
             {
                 var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
                 MakeDeclarationNullable(editor, node, alreadyHandled);
             }
 
-            alreadyHandled.Free();
```

Comment:
these are generally bugs as we might not reach here if we return early (i.e. if cancellationToken triggers).